pkg/upgrade/upgrade.go

```diff
@@ -688,3 +691,44 @@ func RemoveLabel(cli client.Client, objectName string, labelKey string) error {
 	}
 	return nil
 }
+
+func deleteDeprecatedNamespace(ctx context.Context, cli client.Client, namespace string) error {
+	foundNamespace := &corev1.Namespace{}
+	if err := cli.Get(ctx, client.ObjectKey{Name: namespace}, foundNamespace); err != nil {
+		if apierrs.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("could not get %s namespace: %w", namespace, err)
+	}
+
+	// Check if namespace is owned by DSC
+	isOwnedByDSC := false
+	for _, owner := range foundNamespace.OwnerReferences {
+		if owner.Kind == "DataScienceCluster" {
+			isOwnedByDSC = true
+		}
+	}
+	if !isOwnedByDSC {
+		return nil
+	}
+
+	// Check if namespace has pods running
```

Comment:
We do care about pods only, not configmaps and so on?

Reply:
wont be able to know what exactly can be created in the namespace, pod is the minimum we can check to indicate if this namespace is being used or not